src/MagnumPlugins/UfbxImporter/UfbxImporter.cpp

```diff
@@ -1818,20 +1823,23 @@ Containers::Optional<AnimationData> UfbxImporter::doAnimation(UnsignedInt id) {
         while(valueBuffer.size() % valueAlignment != 0)
             arrayAppend(valueBuffer, 0);
         std::size_t valueOffset = valueBuffer.size();
+        std::size_t valueElementSize = 0;
 
         switch(target) {
-        case AnimationTrackTargetType::Translation3D:
+        case AnimationTrackTarget::Translation3D:
             {
                 Containers::ArrayView<Vector3> values = Containers::arrayCast<Vector3>(arrayAppend(valueBuffer, NoInit, keyCount * sizeof(Vector3)));
+                valueElementSize = sizeof(Vector3);
```

Comment:
The split seems like a good idea! Computing the values is independent from sampling the times after all, let me see how it would look like!

Reply:
> I'll add a animationTrackTypeSize() query for it so you don't need to store that separately.

For the value array allocation I'd need the alignment as I think I'd store one `StridedArrayView2D<char>` per track using `ArrayTuple::Item(Corrade::NoInitT, std::size_t size, std::size_t elementSize, std::size_t elementAlignment, StridedArrayView2D<char>& outputView);`. Otherwise I would have to have something like an `union` in the track for possible value types and that feels wrong.

Reply:
Yes that would get rid of all the manual sizing, this refactor is turning out incredible btw, love the ability to cast from type-erased `StridedArrayView2D` to typed 1D view, feels so much safer :D

Reply:
This is now implemented -- mosra/magnum@bc1859f653e4dde58df1a1291cd75ce0325bcf30.